reana_client/cli/secrets.py

```diff
@@ -0,0 +1,196 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2019 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+"""REANA client workflow related commands."""
+import base64
+import logging
+import sys
+import traceback
+from builtins import FileNotFoundError
+
+import click
+import tablib
+from reana_client.api.client import add_secrets, delete_secrets, \
+    list_secrets, current_rs_api_client
+from reana_client.cli.utils import add_access_token_options
+from reana_client.config import ERROR_MESSAGES
+from reana_commons.errors import MissingAPIClientConfiguration
+from reana_commons.utils import click_table_printer
+
+
+@click.group(help='Secret management commands')
+@click.pass_context
+def secrets_group(ctx):
+    """Top level wrapper for secrets management."""
+    logging.debug(ctx.info_name)
+
+
+@secrets_group.command()
+@click.option(
+    '--from-literal',
+    multiple=True,
+    help='Secrets to be uploaded from literal string.'
+         'e.g. PASSWORD=password123')
+@click.option(
+    '--from-file',
+    multiple=True,
+    help='Secret to be uploaded from file.')
+@click.option(
+    '--overwrite',
+    is_flag=True,
+    default=False,
+    help='Overwrite the secret if already present')
+@add_access_token_options
+def secrets_add(from_literal, from_file, overwrite, access_token):  # noqa: D301
+    """Add secrets from literal string or from file.
+
+    Examples: \n
+    \t $ reana-client secrets-add --from-literal PASSWORD=password \n
+    \t $ reana-client secrets-add --from-file ~/.keytab \n
+    \t $ reana-client secrets-add --from-literal USER=reanauser \n
+    \t                            --from-literal PASSWORD=password \n
+    \t                            --from-file ~/.keytab
+    """
+    if not access_token:
+        click.echo(
+            click.style(ERROR_MESSAGES['missing_access_token'], fg='red'),
+            err=True)
+        sys.exit(1)
+
+    secrets = {}
+    for literal in from_literal:
```

Comment:
What do you think about extracting all this code to parse secrets from CLI to a function? It could be a [`click.ParamType`](https://click.palletsprojects.com/en/7.x/api/#click.ParamType) with the logic inside [`convert`](https://click.palletsprojects.com/en/7.x/api/#click.ParamType.convert).